workflow/rules/trim.smk

```diff
@@ -13,15 +13,75 @@ rule cutadapt_pe:
         "0.31.1/bio/cutadapt/pe"
 
 
-rule cutadapt:
+if config["experiment"]["3-prime-rna-seq"]["activate"]:
+    
+    
+    rule cutadapt1:
+        input:
+            get_fastqs,
+        output:
+            fastq="results/trimmed/{sample}-{unit}.1.1.fastq.gz",
+            qc="results/trimmed/{sample}-{unit}.1.1.qc.txt",
+        params:
+            extra=lambda w: str( "-m 20 -O 20 -a ""polyA=A{20}"" -a ""QUALITY=G{20}"" -n 2")
+        log:
+            "results/logs/cutadapt/{sample}-{unit}.1.1.log",
+        wrapper:
+            "v1.14.1/bio/cutadapt/se"
+    
+    
+    rule cutadapt2:
+        input:
+            fastq="results/trimmed/{sample}-{unit}.1.1.fastq.gz",
+        output:
+            fastq="results/trimmed/{sample}-{unit}.2.1.fastq.gz",
+            qc="results/trimmed/{sample}-{unit}.2.1.qc.txt",
+        params:
+            adapters=lambda w: str("-m 20 -O 3 --nextseq-trim=10 -a \"""A{18}AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC;min_overlap=3;max_error_rate=0.100000\""""),
+        log:
+            "results/logs/cutadapt/{sample}-{unit}.2.1.log",
+        wrapper:
+            "v1.14.1/bio/cutadapt/se"
+    
+    
+    rule cutadapt3:
+        input:
+            fastq="results/trimmed/{sample}-{unit}.2.1.fastq.gz",
+        output:
+            fastq="results/trimmed/{sample}-{unit}.fastq.gz",
+            qc="results/trimmed/{sample}-{unit}.qc.txt",
+        params:
+            adapters=lambda w: str("-m 20 -O 20 -g \"""r1adapter=AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC;min_overlap=20\""" --discard-trimmed"),
+        log:
+            "results/logs/cutadapt/{sample}-{unit}.log",
+        wrapper:
+            "v1.14.1/bio/cutadapt/se"
```

Comment:
For these 3 steps, I need a detailed explanation, e.g.
1. Why do we trim twice with the same adapter, but different overlaps?
2. The first iteration trims only poly-A and something with the quality? Can't this be done along with any of the other iterations?

In any case, a comment with a rationale and human readable description has to be provided for each iteration.

Reply:
I added the adapter options based on the Lexogen documentation. 
https://faqs.lexogen.com/faq/what-is-the-adapter-sequence-i-need-to-use-for-t-1
As I understood, they have used 3 Cutadapt options without specifying what each iteration does. However, I will add the information based on the options used by referring to the Cutadapt manual.